pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -140,3 +144,9 @@ func NewAuthInfo() *AuthInfo {
 func NewPreferences() *Preferences {
 	return &Preferences{Extensions: make(map[string]runtime.Object)}
 }
+
+// AuthProviderConfig
+type AuthProviderConfig struct {
+	Name   string           `json:"name"`
+	Config *json.RawMessage `json:"config"`
```

Comment:
I'm more used to seeing `runtime.RawExtension`, and `runtime.Object`.  It's what we we use downstream and it works reasonably well.

I do like this a lot better than just a name.